app/models/http_proxy.rb

```diff
@@ -31,8 +33,8 @@ class HttpProxy < ApplicationRecord
 
   def full_url
     uri = URI(url)
-    uri.user = username if username.present?
-    uri.password = password if username.present?
```

Comment:
I wonder about this and if you can set a password if there is no password. So I read [RFC1738](https://www.ietf.org/rfc/rfc1738.txt):
> Note that an empty user name or password is different than no user name or password; there is no way to specify a password without specifying a user name

That makes me think the old code was correct.

Reply:
Now that you point it out I must admit the new behavior is not better, I'll change that. However if a user provides a password without a username, shouldn't it fail hard instead of silently discarding the password?

Reply:
Yes, but perhaps that should already be done in the model validation. I think `full_url` can also be called at runtime and if you would have an invalid model, the question is whether it should hard fail or at least not send anything incorrect.

Perhaps it would be more explicit that this is intentional if it's written as:
```ruby
if username.present?
  uri.user = username
  uri.password = password
end
```